src/imf_reader/sdr/read_announcements.py

```diff
@@ -94,16 +95,31 @@ def get_latest_allocations_holdings_date() -> tuple[int, int]:
 
 
 def fetch_allocations_holdings(date: tuple[int, int] | None = None) -> pd.DataFrame:
-    """Fetch SDR holdings and allocations data for a given date
+    """
+    Fetch SDR holdings and allocations data for a given date. If date is not specified or exceeds the latest available
+    date, it fetches data for the latest date
 
     Args:
-        date: The year and month to get allocations and holdings data for. e.g. (2024, 11) for November 2024. If None, the latest announcements released are fetched
+        date: The year and month to get allocations and holdings data for. e.g. (2024, 11) for November 2024.
+        If None, the latest announcements released are fetched
 
     returns:
         A dataframe with the SDR allocations and holdings data
     """
 
     if date is None:
-        date = get_latest_allocations_holdings_date()
+        date = fetch_latest_allocations_holdings_date()
+    else:
+        # Temporarily disable logging while calling fetch_latest_allocations_holdings_date()
+        logger.setLevel(logging.WARNING)
+        latest_date = fetch_latest_allocations_holdings_date()
+        logger.setLevel(logging.INFO)
```

Comment:
This approach works. The only consideration is that you are hard coding the level back to `INFO` which is not the safest approach. You should rather catch what the original level was - something like `original_level = logger.level` and after completion of the task, set it back to `original_level`